src/components/Navigation/Navigation.jsx

```diff
@@ -26,93 +26,27 @@ import StackOverflowIcon from '../../styles/icons/stack-overflow.svg';
-    toggleSidebar: PropTypes.func,
-    theme: PropTypes.string,
-    switchTheme: PropTypes.func,
-  };
-  render() {
-    const { pathname, links, toggleSidebar, theme, switchTheme } = this.props;
-    const themeSwitcher = () => switchTheme(theme === DARK ? LIGHT : DARK);
+Navigation.propTypes = {
+  pathname: PropTypes.string,
+  hash: PropTypes.string,
+  links: PropTypes.array,
+  toggleSidebar: PropTypes.func,
+  theme: PropTypes.string,
+  switchTheme: PropTypes.func,
+};
 
-    return (
-      <Banner
-        onSearch={onSearch}
-        blockName="navigation"
-        logo={<Logo light={true} />}
-        url={pathname}
-        items={[
-          ...links,
-          {
-            title: 'GitHub Repository',
-            url: 'https://github.com/webpack/webpack',
-            className: 'navigation__item--icon',
-            content: <GithubIcon aria-hidden="true" fill="#fff" width={16} />,
-          },
-          {
-            title: 'webpack on Twitter',
-            url: 'https://twitter.com/webpack',
-            className: 'navigation__item--icon',
-            content: <TwitterIcon aria-hidden="true" fill="#fff" width={16} />,
-          },
-          {
-            title: 'webpack on Stack Overflow',
-            url: 'https://stackoverflow.com/questions/tagged/webpack',
-            className: 'navigation__item--icon',
-            content: (
-              <StackOverflowIcon aria-hidden="true" fill="#fff" width={16} />
-            ),
-          },
-          {
-            className: 'navigation__item--icon',
-            content: (
-              <Dropdown
-                className="navigation__languages"
-                items={[
-                  {
-                    title: 'English',
-                    url: 'https://webpack.js.org/',
-                  },
-                  {
-                    lang: 'zh',
-                    title: '中文',
-                    url: 'https://webpack.docschina.org/',
-                  },
-                  {
-                    lang: 'ko',
-                    title: '한국어',
-                    url: 'https://webpack.kr/',
-                  },
-                ]}
-              />
-            ),
-          },
-          {
-            className: 'navigation__item--icon',
-            content: (
-              <button
-                style={{
-                  background: 'transparent',
-                  border: 'none',
-                  cursor: 'pointer',
-                }}
-                onClick={themeSwitcher}
-              >
-                {theme === DARK ? '🌙' : '☀️'}
-              </button>
-            ),
-          },
-        ]}
-        link={Link}
-        onMenuClick={toggleSidebar}
-      />
-    );
-  }
+function Navigation({
+  pathname,
+  hash = '',
+  links,
+  toggleSidebar,
+  theme,
+  switchTheme,
+}) {
+  const themeSwitcher = () => switchTheme(theme === DARK ? LIGHT : DARK);
+  const [locationHash, setLocationHash] = useState('');
```

Comment:
```suggestion
  const [locationHash, setLocationHash] = useState(hash);
```
Can we initialize it with a default value from prop? Any possible flaw?

Reply:
I was afraid of `undefined` as a string. but actually, that's not going to happen. I'll fix it